### PR TITLE
Annotate Cmm.Pop with the exception handler label

### PR DESCRIPTION
**Pull request by mshinwell**
As per request in #1380 

cc @xclerc does this work?  I had to add a "generic pop" case as sometimes there isn't a handler label; is this ok?

### CODE REVIEW

**Comment by gretay-js** [comment]
> sometimes there isn't a handler label 

do you mean that sometimes the label is not known statically?

**Comment by mshinwell** [comment]
> > sometimes there isn't a handler label
> 
> do you mean that sometimes the label is not known statically?

It arises for non-delayed try-with.

**Review by Gbury — APPROVED** [review]
This is good for me, but @lthls would probably have a more interesting opinion on the design.

**Comment by gretay-js** [comment]
Ah, if `Pop_generic` is only introduced in `cfgize` for non-delayed handlers, then we probably know the label ther already, but don't yet keep track of it.  Should be easy to fix `State.add_iend_with_poptrap` to take the label, it should be `label_handler` in L590.

**Comment by xclerc** [comment]
I think I am a bit worried about the `Pop_generic`
in `Selectgen`. Is it the case that such a pop can
be used to close one of several possible pushes?

**Comment by xclerc** [comment]
Ah, yes, sorry - I now understand the "sometimes there isn't a handler label"
part of Mark's message above. Then, that should work: on the CFG, when an
exception handler is dead, we can now delete it. What was stopping us before
was that we needed not only to remove the dead nodes but also the `Pushtrap`
and `Poptrap` referencing them.  After this PR, that seems trivial to do. thanks!

**Comment by mshinwell** [comment]
Since this is for Cfg I'm not sure we need to port it to `ocaml/`.  I've fixed the failing testcase.

**Comment by mshinwell** [comment]
@xclerc please approve

File: backend/cmmgen.ml [hunk 1]
```diff
@@ -106,6 +106,13 @@ let enter_catch_body env nfail =
   }
 
 let mk_traps env nfail =
+  let catch_trywith_depths_inverse =
+    env.catch_trywith_depths
+    |> IntMap.bindings
+    |> List.map (fun (nfail, depth) -> depth, nfail)
+    |> List.to_seq
```
Comment (xclerc) [inline backend/cmmgen.ml:113]:
```suggestion
    |> IntMap.to_seq
    |> Seq.map (fun (nfail, depth) -> depth, nfail)
```
(untested, and quite possibly not worth the trouble)

**Review by xclerc — APPROVED** [review]

**Pull request merged**